.github/workflows/go.yml

```diff
@@ -24,3 +24,5 @@ jobs:
     - uses: dominikh/staticcheck-action@v1.3.0
       with:
         version: "2023.1"
+        # go is already installed by actions/setup-go with the correct version from go.mod
```

Comment:
I attempt to fix the build by adjusting the version of golang used during this step.

As per https://staticcheck.dev/docs/running-staticcheck/ci/github-actions/#managing-go
We should rely on `actions/setup-go@v5` to install go with the correct version from go.mod which I've also adjusted.

The build was failing because the version of go installed during this step was 1.19

Reply:
Didn't know about `go-version-file`, thanks. Will use this for our internal configs too. 👌